pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -591,3 +592,65 @@ func getContainerResources(container v1.Container, replicas float64) *clustersta
 	req.GPU = req.GPU * int64(replicas)
 	return req
 }
+
+//return value is a string for flexibility of expression
+func (gr *GenericResources) GetGenericItemKindStatus(aw *arbv1.AppWrapperGenericResource, namespace string) (completed bool) {
+	dd := gr.clients.Discovery()
+	apigroups, err := restmapper.GetAPIGroupResources(dd)
+	if err != nil {
+		klog.Errorf("Error getting API resources, err=%#v", err)
+	}
+	restmapper := restmapper.NewDiscoveryRESTMapper(apigroups)
+	_, gvk, err := unstructured.UnstructuredJSONScheme.Decode(aw.GenericTemplate.Raw, nil, nil)
+	if err != nil {
+		klog.Errorf("Decoding error, please check your CR! Aborting handling the resource creation, err:  `%v`", err)
+	}
+	//service and podgroup has no conditions return and ignore
+	//TODO: make it configurable?
+	if gvk.GroupKind().Kind == "Service" || gvk.GroupKind().Kind == "PodGroup" {
+		return false
+	}
+
+	mapping, err := restmapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		klog.Errorf("mapping error from raw object: `%v`", err)
+	}
+	restconfig := gr.kubeClientConfig
+	restconfig.GroupVersion = &schema.GroupVersion{
+		Group:   mapping.GroupVersionKind.Group,
+		Version: mapping.GroupVersionKind.Version,
+	}
+	rsrc := mapping.Resource
+	dclient, err := dynamic.NewForConfig(restconfig)
+	if err != nil {
+		klog.Errorf("Error creating new dynamic client, err=%#v", err)
+	}
+
+	inEtcd, err := dclient.Resource(rsrc).Namespace(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Error listing object: ", err)
+	}
+
+	for _, job := range inEtcd.Items {
+		completionRequiredBlock := aw.CompletionStatus
+		if len(completionRequiredBlock) > 0 {
+			conditions, err := job.Object["status"].(map[string]interface{})["conditions"].([]interface{})
+			//if condition not found skip for this interation
+			if err {
+				klog.Errorf("Error processing unstructured object")
```

Comment:
This is actually an error from the user which should be reflected in the AW conditions.  For now let's continue to log as an error and add a `TODO:...`

Reply:
Done, thanks